timemachine/kernels/wrap_kernels.cpp

```diff
@@ -10,12 +13,135 @@
 
 namespace py = pybind11;
 
+template <typename RealType>
+void declare_context(py::module &m, const char *typestr) {
+
+    using Class = timemachine::Context<RealType>;
+    std::string pyclass_name = std::string("Context_") + typestr;
+    py::class_<Class>(
+        m,
+        pyclass_name.c_str(),
+        py::buffer_protocol(),
+        py::dynamic_attr()
+    )
+    .def(py::init([](
+        const std::vector<timemachine::Potential<RealType> *> system,
```

Comment:
@bp-kelley these two pointers are the two confusing ones, Potential is a pure abstract class whose concrete subclasses are instantiated and kept around on the python side, so I need a pointer here. However, I absolutely do *not* want to transfer over ownership to the Context class since I need them to be alive after the Context Destructor is called (which is how it's currently done).